### PR TITLE
nixos/nginx: Allow empty port for listen directive (for unix socket)

### PR DESCRIPTION
**Pull request by dongcarl**
When listening on unix sockets, it doesn't make sense to specify a port for nginx's listen directive.

Since nginx defaults to port 80 when the port isn't specified (but the address is), we can change the default for the option to null as well without changing any behaviour.

### CODE REVIEW

**Comment by RaitoBezarius** [comment]
Please write a NixOS test for testing this NGINX scenario, while the change looks pretty trivial, we need long term maintenance of such things.

**Comment by dongcarl** [comment]
> Please write a NixOS test for testing this NGINX scenario, while the change looks pretty trivial, we need long term maintenance of such things.

Test added.

**Comment by RaitoBezarius** [comment]
Looks nice to me, I will defer to @tfc for the test driver bits.

**Review by RaitoBezarius — APPROVED** [review]
For the NGINX bits, good to go.

**Comment by SuperSandro2000** [comment]
> Since nginx defaults to port 80 when the port isn't specified (but the address is), we can change the default for the option to null as well without changing any behaviour.

I would still keep the default of 80 because it is more obvious for people reading the code.

File: nixos/modules/services/web-servers/nginx/vhost-options.nix
Comment (SuperSandro2000) [inline nixos/modules/services/web-servers/nginx/vhost-options.nix:39]:
```suggestion
            default = 80;
```

**Review by tfc — APPROVED** [review]

File: nixos/modules/services/web-servers/nginx/vhost-options.nix
Comment (RaitoBezarius) [reply nixos/modules/services/web-servers/nginx/vhost-options.nix:39]: I have no strong opinion on this one, it's true that the default is more readable this way, or the description should reflect that the default is 80 because of NGINX. As you wish @dongcarl.

Comment (dongcarl) [reply nixos/modules/services/web-servers/nginx/vhost-options.nix:39]: I might be mistaken, but if the default is `80`, then wouldn't unix socket users need to explicitly specify this?
```
{
  addr = "unix:/var/run/nginx.sock";
  port = null;
}
```

I'm happy to add on to the description if people feel strongly about it, but not sure that we should in case `nginx`'s behavior for the default changes in the future.

**Comment by dongcarl** [comment]
Please do not merge this even though there are approvals, there are bugs. Fixing now

File: nixos/modules/services/web-servers/nginx/vhost-options.nix
Comment (RaitoBezarius) [reply nixos/modules/services/web-servers/nginx/vhost-options.nix:39]: Hm... I think we can go with the current state and see if we need to adjust this later on.

Comment (SuperSandro2000) [reply nixos/modules/services/web-servers/nginx/vhost-options.nix:39]: This should have been at least mentioned in the description. I expect most people to be a bit confused about this now.

**Comment by SuperSandro2000** [comment]
Please take a look at #262796

**Pull request merged**
